Make colony home page mobile responsive

willm30 opened the pull request:
## Description

This PR is the feature for a mobile responsive Colony Home page. It accords with Screen 1 of the [Figma design](https://www.figma.com/file/DjVCPq7LpkjEMEfQZtKgBdC1/Style-Guide?node-id=6641%3A9676). 

Note: it does not handle the wiring of the hamburger menu or the rewiring of the user avatar (colony summary) popover menu (i.e. screens 3 and 7). It also does not deal with any of the other screens. These will all be included in further PRs. It's purely focused on the layout of the Colony home page. 

**New stuff** ✨

- [ ] #3486 
- [ ] #3495
- [ ] #3499

## Screenshots 

**_Signed In_**
![colony-home](https://user-images.githubusercontent.com/64402732/174488799-4213c9f0-1af1-446a-9ae1-10ea98160817.png)

**_Signed Out_**
![colony-home-signout](https://user-images.githubusercontent.com/64402732/174488923-c452ec5d-89f9-4991-a651-d53b682bdc35.png)


## TODO

Sections To Complete:

- [x] Colony Avatars / User Avatar / Hamburger Menu
- [x] Colony Info / Join Colony
  - [x] Align colony info popover 
- [x] Funds info / manage funds
  - [x] Align coin info popover 
- [x] Actions menu
  - [x] Align create new team menu    
  - [x] Align new action menu
- [x] Actions summary
  - [x] Add load more button for actions
- [x] Connect wallet (signed out)
  - [x] Align button 
  - [x] Adjust menu

Resolves: #3475 
Resolves: #3486 
Resolves: #3495
Resolves: #3499

dit7ya commented:
This is huge, whoa! Are we sure there is no other simpler way to achieve this without external libraries? (It might very well be the case but I am just double-checking).

willm30 commented:
> This is huge, whoa! Are we sure there is no other simpler way to achieve this without external libraries? (It might very well be the case but I am just double-checking).

I went for [react-responsive](https://github.com/yocontra/react-responsive) mainly for peace of mind. We could write our own hook to detect screen size I guess but using the library was quicker and probably more reliable short-term (better tested).

I guess we generally try to avoid external libraries where possible?

rdig commented on src/modules/pages/components/RouteLayouts/Default/Default.tsx:
```diff
@@ -45,10 +51,24 @@ const Default = ({
-          {hasSubscribedColonies && (
-            <div className={styles.coloniesList}>
-              <SubscribedColoniesList />
+          {isMobile ? (
+            // Render the UserNavigation and SubscribedColoniesList in shared parent on mobile
+            <div className={styles.head}>
+              <div className={navStyles.nav}>
+                <UserNavigation />
+              </div>
+              {hasSubscribedColonies && (
+                <div className={styles.coloniesList}>
+                  <SubscribedColoniesList />
+                </div>
+              )}
             </div>
+          ) : (
+            hasSubscribedColonies && (
+              <div className={styles.coloniesList}>
+                <SubscribedColoniesList />
+              </div>
+            )
```
Please try to reduce code repetition as much as possible:
```suggestion
const SubscribedColonies = () => hasSubscribedColonies ? (
    <div className={styles.coloniesList}>
      <SubscribedColoniesList />
    </div>
  ) : null;

{isMobile ? (
  // Render the UserNavigation and SubscribedColoniesList in shared parent on mobile
  <div className={styles.head}>
    <div className={navStyles.nav}>
      <UserNavigation />
    </div>
    {SubscribedColonies()}
  </div>
) : SubscribedColonies()
```

rdig reviewed (changes requested):
This is mostly good and kudos to you for the amount of work you've put in. :100: 

Other than what I've picked up in the review _(see comments)_, there's also some other things need discussing:

1. We've changed the design specs for the _Joined Colonies_ list, so your implementation is no longer up to spec. You can tackle it here, or in a separate PR, up to you.

2. There's a discussion on how tread intermediate view port sizes, and if we should cater to them _(ie: tablets / smaller laptops)_, since the view is good for vertical view phones, >1200 desktop sizes, but broken for everything in between.

So I'd suggest the 3 of us _(me, you, Arren)_ get together in a call an hash this out.

willm30 commented on src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.tsx:
```diff
@@ -81,13 +87,14 @@ const ColonySubscription = ({
           onClick={() => subscribe()}
           appearance={{ theme: 'blue', size: 'small' }}
           data-test="joinColonyButton"
+          className={styles.joinButton}
```
This I saw after I submitted the PR and addressed it in the other PR I submitted. Am I ok to leave it in the other one?

willm30 commented:
> 1. We've changed the design specs for the _Joined Colonies_ list, so your implementation is no longer up to spec. You can tackle it here, or in a separate PR, up to you.

I will address it in a separate PR.

willm30 commented:
@rdig I have implemented the changes requested, ready for a re-review and updated the PR body (inc updated screenshots). Will split out screen 2 (joined colonies list) into separate PR.

arrenv reviewed (approved):
It is looking good to me, just a couple of comments.

willm30 commented on src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.tsx:
```diff
@@ -87,7 +87,7 @@ const ColonySubscription = ({
       {!isSubscribed && !username && (
         <Link
-          className={styles.createUserRedirect}
+          className={`${styles.createUserRedirect}`}
```
Good spot

rdig reviewed (approved):
All good now. Nice work Will! :tada:

rdig commented:
Also @willm30 as per Github's specs liked issues should be one per line.

So this:
![Screenshot from 2022-06-21 23-05-44](https://user-images.githubusercontent.com/1193222/174888196-6d06275d-a880-48d8-85ba-97fa3f4644ce.png)

Should be written as:
```
Resolves #3475
Resolves #3486
Resolves #3495
Resolves #3499
```

See more: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
